### PR TITLE
Move most ui-fulldeps tests to ui/

### PR DESCRIPTION
**Pull request by jyn514**
Same rationale as https://github.com/rust-lang/rust/pull/109770, they don't actually need a stage 2 build.

This increases the limit for the UI directory because otherwise it was annoying to be constantly moving files into subdirectories when I fixed a test; https://github.com/rust-lang/rust/pull/109873 makes up for it.

cc https://github.com/rust-lang/rust/pull/109770, https://github.com/rust-lang/rust/pull/109874

### CODE REVIEW

**Comment by jyn514** [comment]
Ok, this is finally ready for review - sorry for the noise while I got CI working, didn't have access to my fast computer the last couple weeks.

**Comment by jyn514** [comment]
I've fixed the tests. One of them is a real bug on wasm32 which I'm ignoring for now, another is a `std` test that needs an OS, the third one needed some git archeology. Would appreciate a review on the third test to make sure you still think it's testing the original problem.

File: tests/ui/extern-mod-syntax.rs [hunk 1]
```diff
@@ -1,11 +1,11 @@
 // run-pass
```
Comment (compiler-errors) [inline tests/ui/extern-mod-syntax.rs:1]: :+1:

File: tests/ui/issue-13560.rs [hunk 2]
```diff
@@ -1,4 +1,5 @@
 // run-pass
+// ignore-wasm32-unknown-unknown (apparently the compiler still has the original bug there??)
```
Comment (compiler-errors) [inline tests/ui/issue-13560.rs:2]: i wonder if we should file a bug for this..?

File: tests/ui/std/switch-stdout.rs [hunk 3]
```diff
@@ -1,4 +1,5 @@
 // run-pass
+// ignore-wasm32-unknown-unknown (needs file descriptors)
```
Comment (compiler-errors) [inline tests/ui/std/switch-stdout.rs:2]: :+1:

**Review by compiler-errors — COMMENTED** [review]
r=me either way

File: tests/ui/issue-13560.rs (hunk 2 above)
Comment (jyn514) [reply tests/ui/issue-13560.rs:2]: Good idea, done: https://github.com/rust-lang/rust/issues/110172

**Comment by jyn514** [comment]
I'm going to disable `issue-13560.rs` when cross-compiling for now. It needs something like a `// needs-crate-type-dynamic` header, which doesn't exist today, and the fulldeps tests previously weren't cross-compiled anyway.

**Pull request merged**